conductor/src/main.rs

```diff
@@ -40,36 +41,54 @@ const REQUEUE_VT_SEC_SHORT: i32 = 5;
-        env::var("METRICS_EVENTS_QUEUE").expect("METRICS_EVENTS_QUEUE must be set");
-    let data_plane_events_queue =
-        env::var("DATA_PLANE_EVENTS_QUEUE").expect("DATA_PLANE_EVENTS_QUEUE must be set");
-    let data_plane_basedomain =
-        env::var("DATA_PLANE_BASEDOMAIN").expect("DATA_PLANE_BASEDOMAIN must be set");
-    let backup_archive_bucket =
-        env::var("BACKUP_ARCHIVE_BUCKET").expect("BACKUP_ARCHIVE_BUCKET must be set");
-    let storage_archive_bucket =
-        env::var("STORAGE_ARCHIVE_BUCKET").expect("STORAGE_ARCHIVE_BUCKET must be set");
-    let cf_template_bucket =
-        env::var("CF_TEMPLATE_BUCKET").expect("CF_TEMPLATE_BUCKET must be set");
-    let max_read_ct: i32 = env::var("MAX_READ_CT")
-        .unwrap_or_else(|_| "100".to_owned())
+    let pg_conn_url = get_env_var("POSTGRES_QUEUE_CONNECTION", None)?;
+    let control_plane_events_queue = get_env_var("CONTROL_PLANE_EVENTS_QUEUE", None)?;
+    let metrics_events_queue = get_env_var("METRICS_EVENTS_QUEUE", None)?;
+    let data_plane_events_queue = get_env_var("DATA_PLANE_EVENTS_QUEUE", None)?;
+    let data_plane_basedomain = get_env_var("DATA_PLANE_BASEDOMAIN", None)?;
+    let backup_archive_bucket = get_env_var("BACKUP_ARCHIVE_BUCKET", None)?;
+    let storage_archive_bucket = get_env_var("STORAGE_ARCHIVE_BUCKET", None)?;
+    let cf_template_bucket = get_env_var("CF_TEMPLATE_BUCKET", None)?;
+
+    let max_read_ct: i32 = get_env_var("MAX_READ_CT", Some("100"))?
         .parse()
-        .expect("error parsing MAX_READ_CT");
-    let is_cloud_formation: bool = env::var("IS_CLOUD_FORMATION")
-        .unwrap_or_else(|_| "true".to_owned())
-        .parse()
-        .expect("error parsing IS_CLOUD_FORMATION");
-    let aws_region: String = env::var("AWS_REGION")
-        .unwrap_or_else(|_| "us-east-1".to_owned())
+        .map_err(|e| {
+            ConductorError::ConnectionPoolError(format!("Failed to parse MAX_READ_CT: {}", e))
+        })?;
+    let is_cloud_formation: bool = get_env_var("IS_CLOUD_FORMATION", Some("true"))?
         .parse()
-        .expect("error parsing AWS_REGION");
+        .map_err(|e| {
+            ConductorError::ConnectionPoolError(format!(
+                "Failed to parse IS_CLOUD_FORMATION: {}",
+                e
+            ))
+        })?;
+    let aws_region = get_env_var("AWS_REGION", Some("us-east-1"))?;
+    let is_gcp: bool = get_env_var("IS_GCP", Some("false"))?.parse().map_err(|e| {
+        ConductorError::ConnectionPoolError(format!("Failed to parse IS_GCP: {}", e))
+    })?;
+    let gcp_project_id = get_env_var("GCP_PROJECT_ID", None)?;
+    let gcp_project_number = get_env_var("GCP_PROJECT_NUMBER", None)?;
+
+    // Error and exit if both IS_CLOUD_FORMATION and IS_GCP are set to true
+    if is_cloud_formation && is_gcp {
+        return Err(ConductorError::ConnectionPoolError(
+            "Cannot have both IS_CLOUD_FORMATION and IS_GCP set to true".into(),
+        ));
+    }
```

Comment:
in cases of invalid configuration, just crash the program with the error, so it's impossible to start it